src/Ashampoo.Translation.Systems.Formats/src/QT/QTFormat.cs

```diff
@@ -0,0 +1,177 @@
+            options!.IsCancelled = true;
+            return;
+        }
+
+        Guard.IsNotNull(Header.TargetLanguage.Value);
+
+        var doc = XDocument.Load(stream);
+        var messageElements = doc.Descendants().Where(e => e.Name == "message");
+
+        await ReadTranslations(messageElements);
+    }
+
+    private async Task ReadTranslations(IEnumerable<XElement> elements)
+    {
+        foreach (var element in elements)
+        {
+            if (element.IsEmpty || element.NodeType is XmlNodeType.EndElement)
+            {
+                continue;
+            }
+
+            var sourceElement = element.Element("source");
+            var translationElement = element.Element("translation");
+            var translationType = translationElement?.Attribute("type");
+            var comments = await SetComments(element);
+
+            if (!string.IsNullOrWhiteSpace(sourceElement?.Value))
+            {
+                var unit = new DefaultTranslationUnit(sourceElement.Value);
+                var translation = new QtTranslationString(translationElement?.Value ?? string.Empty,
+                    Header.TargetLanguage, comments);
+                if (translationType is not null &&
+                    Enum.TryParse<QtTranslationType>(translationType.Value, true, out var result))
+                {
+                    translation.Type = result;
+                }
+
+                unit.Translations.Add(translation);
+                TranslationUnits.Add(unit);
+                Console.WriteLine("Add translation with id {0} and value {1}. Comments: {2}", unit.Id,
+                    translation.Value, string.Join(",", translation.Comments));
+            }
+        }
+    }
+
+    private Task<List<string>> SetComments(XElement element)
+    {
+        List<string> comments = [];
+        if (element.TryGetElement("translatorcomment", out var translatorComment))
+        {
+            if (translatorComment?.Value != null) comments.Add(translatorComment.Value);
+        }
+
+        if (element.TryGetElement("extracomment", out var extraComment))
+        {
+            if (extraComment?.Value != null) comments.Add(extraComment.Value);
+        }
+
+        return Task.FromResult(comments);
+    }
+
+    /// <inheritdoc />
+    public async Task WriteAsync(Stream stream)
+    {
+        var layout = $"""
+                      <?xml version="1.0" encoding="utf-8"?>
+                      <!DOCTYPE TS>
+                      <TS version="2.0" language="{Header.TargetLanguage.Value.Replace('-', '_')}">
+                      </TS>
+                      """;
+
+        var doc = XDocument.Parse(layout);
+        var tsElement = doc.Descendants().First(e => e.Name == "TS");
+        var writer = tsElement.CreateWriter();
+        await WriteTranslations(writer);
+        doc.Save(stream);
+    }
+
+    [SuppressMessage("ReSharper", "MethodHasAsyncOverload")]
+    private async Task WriteTranslations(XmlWriter writer)
```

Comment:
Der Writer unterstüzt asnychrones Schreiben

Reply:
Die thrown NotImplementedException. Durch den Aufruf `var writer = tsElement.CreateWriter();` in der WriteAsync Methode kriege ich nur einen Synchronen Writer zurück. Ich habe keine Möglichkeit vom XDocument einen Asynchronen Writer zu bekommen.